R/list-combine.R

```diff
@@ -95,3 +105,8 @@ check_list_of_data_frames <- function(x, error_call = caller_env()) {
     call = error_call
   )
 }
+
+## used to convert empty elements into NA for list_binding functions
+convert_empty_element_to_NA = function(x) {
+  map(x, function(x) if(vctrs::vec_is_empty(x)) NA else x)
```

Comment:
This might have quite a performance impact if `x` is big. A faster alternative would be something like

```r
empty <- vctrs::list_sizes(x) == 0
x[empty] <- NA
x
```